repose-aggregator/components/filters/body-extractor-to-header/src/test/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilterTest.scala

```diff
@@ -0,0 +1,452 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.bodyextractortoheader
+
+import java.net.URL
+import java.nio.charset.StandardCharsets
+import javax.servlet.{FilterChain, ServletResponse}
+
+import com.mockrunner.mock.web.MockFilterConfig
+import org.junit.Assert.{assertFalse, assertTrue}
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Matchers._
+import org.mockito.Mockito._
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.bodyextractortoheader.config.{BodyExtractorToHeaderConfig, Extractor}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+import org.springframework.mock.web.{MockHttpServletRequest, MockHttpServletResponse}
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[JUnitRunner])
+class BodyExtractorToHeaderFilterTest extends FunSpec with BeforeAndAfter with Matchers {
+
+  val mockConfigurationService = mock(classOf[ConfigurationService])
+  val mockFilterConfig = new MockFilterConfig
+  var config: BodyExtractorToHeaderConfig = _
```

Comment:
Mixing val and var declarations isn't my favorite. I generally do a block of vals followed by a block of vars to keep things straight. That might be a personal preference thing.